mat/cholesky.go

```diff
@@ -697,3 +701,132 @@ func (c *Cholesky) SymRankOne(orig *Cholesky, alpha float64, x Vector) (ok bool)
 func (c *Cholesky) valid() bool {
 	return c.chol != nil && !c.chol.IsEmpty()
 }
+
+type BandCholesky struct {
+	// The chol pointer must never be retained as a pointer outside the Cholesky
+	// struct, either by returning chol outside the struct or by setting it to
+	// a pointer coming from outside. The same prohibition applies to the data
+	// slice within chol.
+	chol *TriBandDense
+	cond float64
+}
+
+func (c *BandCholesky) Factorize(a SymBanded) (ok bool) {
+	n, k := a.SymBand()
+	if c.chol == nil {
+		c.chol = NewTriBandDense(n, k, Upper, nil)
+	} else {
+		c.chol.Reset()
+		c.chol.ReuseAsTriBand(n, k, Upper)
+	}
+	copySymBandIntoTriBand(c.chol, a)
+	cSym := blas64.SymmetricBand{
+		Uplo:   blas.Upper,
+		N:      n,
+		K:      k,
+		Data:   c.chol.RawTriBand().Data,
+		Stride: c.chol.RawTriBand().Stride,
+	}
+	_, ok = lapack64.Pbtrf(cSym)
+	if !ok {
+		c.Reset()
+		return false
+	}
+	work := getFloats(3*n, false)
+	iwork := getInts(n, false)
+	aNorm := glapack.Implementation{}.Dlansb(CondNorm, cSym.Uplo, cSym.N, cSym.K, cSym.Data, cSym.Stride, work)
```

Comment:
Would it be worth adding `Lansb` (and `Lantb`) to lapack64, but not having them use the `lapack64.lapack64` value? Then `Lansb` would look like
```
// Lansb computes the specified norm of an n×n symmetric band matrix. If
// norm == lapack.MaxColumnSum or norm == lapack.MaxRowSum, work must have length
// at least n and this function will panic otherwise.
// There are no restrictions on work for the other matrix norms.
//
// Dlansb is not part of the lapack.Float64 interface and so calls to Lansb are always
// executed by the Gonum implementation.
func Lansb(norm lapack.MatrixNorm, a blas64.SymmetricBand, work []float64) float64 {
	return gonum.Implementation{}.Dlansb(norm, a.Uplo, a.N, a.K, a.Data, max(1, a.Stride), work)
}
```
This way we're not faking it, but we do have the nicer API.

Reply:
Good idea, done.